cast/src/lib.rs

```diff
@@ -443,6 +445,62 @@ where
         Ok(self.provider.get_gas_price().await?)
     }
 
+    pub async fn ls(&self) -> Result<String> {
+        // create a vector of addresses
+        let mut addrs: Vec<H160> = Vec::new();
+
+        // fetch the chain id
+        let chain_id = self.chain_id().await?;
+
+        // for each file in $HOME/.ethereum/keystore
+        if let Ok(home_dir) = std::env::var("HOME") {
+             let path = std::path::Path::new(&home_dir).join(".ethereum").join("keystore");
+             let entries = fs::read_dir(path)?;
+             for entry in entries {
+                 let entry = entry?;
+                 let path = entry.path();
+                 let file = std::io::BufReader::new(std::fs::File::open(&path)?);
+                 let mut value: serde_json::Value = serde_json::from_reader(file)?;
+
+                 let map = value.as_object_mut().unwrap();
+                 let addr = map.get_mut("address").unwrap().as_str().unwrap();
+                 let t = H160::from_str(addr).unwrap();
+                 addrs.push(t);
+             }
+        }
+
+        match Ledger::new(HDPath::LedgerLive(0), chain_id.as_u64()).await {
+            Ok(ledger) => {
+                for i in 0..6 {
+                    let path = HDPath::LedgerLive(i);
+                    let address = ledger.get_address_with_path(&path).await?;
+                    addrs.push(address)
+                }
+                for i in 0..6 {
+                    let path = HDPath::Legacy(i);
+                    let address = ledger.get_address_with_path(&path).await?;
+                    addrs.push(address)
+                }
+
+            }
+            Err(_) => {}
+        }
+
+        let balances = join_all(addrs.iter().map(|addr| self.balance(addr.clone(), None))).await;
+
+        let mut s = String::new();
+        for (i, balance) in balances.iter().enumerate() {
+            let addr = addrs[i];
+            let result = match balance {
+                Ok(balance) => format!("{:?}\t{}\n", addr, balance),
+                Err(_) => format!("{:?}\t{}\n", addr, 0),
+            };
+            s.push_str(&result);
+        }
```

Comment:
I would rather like the function to return the pairs of (Address, Result<Balance>)
and do we do this formatting part outside, when printing to stdout, basically in the `Subcommand::Ls`